### ISSUE TITLE
none async functions behave like async functions

### ISSUE THREAD

**Issue by F3der1co** [Scirra/Construct-bugs#8273]
## Problem description
none async functions behave like async functions

## Attach a .c3p
[waitForPrev.zip](https://github.com/user-attachments/files/17263739/waitForPrev.zip)

## Steps to reproduce

1. run preview

## Observed result
text is set only after tween finished

## Expected result
text is immediately set

## More details
It seems like the asyncronous booleans in the "Edit function" dialog is not working correctly.
Custom actions have the same issue, but I guess that is not a surprise as they share the same implementation.

**Affected browsers/platforms:** 

**First affected release:** 

## System details

<details><summary>View details</summary>

PASTE HERE

</details>

**Comment by F3der1co** [comment]
https://github.com/Scirra/Construct-bugs/issues/5257
Seems to be a duplicate of this, but this definitely goes against how scope and sync functions work for everything else so this is very weird and counter intuitive. It's also confusing as I async functions have an additional overhead according to the documentation, so how does it even happen if the function is not async, are they inlined?

**Comment by F3der1co** [comment]
![image](https://github.com/user-attachments/assets/0c3b8bba-7f40-4843-b64c-7347fefd80f5)
can be worked around by doing this for example, this does add a 1 tick delay before the tween starts though.

**Comment by AshleyScirra** [comment]
I do think this is a duplicate of #5257 so closing, but I will have a look at the older issue and see how feasible it is to fix it. Long-standing issues like this can end up with awkward backwards-compatibility implications though.
